README.md

```diff
@@ -4,32 +4,123 @@
+## Prerequisites
 
-- Intel hardware
-- Fully configured [Kubernetes cluster]
-- Working [Go environment]
+Prerequisites for building and running these device plugins include:
+
+- Appropriate hardware
+- A fully configured [Kubernetes cluster]
+- A working [Go environment]
+
+## Plugins
+
+The below sections detail existing plugins developed using the framework.
+
+### GPU device plugin
+
+The [GPU device plugin](cmd/gpu_plugin/README.md) supports Intel
+[GVT-d](https://github.com/intel/gvt-linux/wiki/GVTd_Setup_Guide) device passthrough
+and acceleration using GPUs of the following hardware families:
+
+- Integrated GPUs within Intel Core processors
+- Intel Xeon processors
+- Intel Visual Compute Accelerator (Intel VCA)
+
+The demo subdirectory contains both a [GPU plugin demo video](demo#intel-gpu-device-plugin-demo-video)
+as well as code for an OpenCL [FFT demo](demo/ubuntu-demo-opencl).
+
+### FPGA device plugin
+
+The [FPGA device plugin](cmd/fpga_plugin/README.md) supports Intel Arria 10 and Stratix 10
+FPGA passthrough. It supports the [OPAE](https://01.org/opae) mode.
+
+The FPGA plugin comes as three parts. Start with the [device plugin](#device-plugin), and then
+examine the [admissing controller](#admission-controller) and the
+[CRIO-O prestart hook](#cri-o-prestart-hook).
+
+> **FIXME** - unclear where the [fpga_tool](cmd/fpga_tool) comes into play. We should document this.
+
+The demo subdirectory contains a [video](demo#intel-fpga-device-plugin-demo-video) showing deployment
+and use of the FPGA plugin. Sources relating to the demo can be found in the
+[opae-nlb-demo](demo/opae-nlb-demo) subdirectory.
+
+#### device plugin
+
+The [FPGA device plugin](cmd/fpga_plugin/README.md) is responsible for discovering and reporting FPGA
+devices to `kubelet`. During the allocation phase, it instructs `kubelet` about device nodes to be
+accessed by the container and sets the annotation that is later on passed to CRI to trigger programming
+by the CRI-O hook.
+
+#### admission controller
+
+The [FPGA admission controller webhook](cmd/fpga_admissionwebhook/README.md) is responsible for performing
+mapping from user-friendly function IDs to the Interface ID and Bitstream ID that are required for FPGA
+programming by the CRI-O hook. Mappings are stored in namespaced custom resource definition (CRD) objects,
+therefore the admission controller also performs access control, determining which bitstream can be used
+for which namespace. The admission controller also keeps the user from bypassing namespaced mapping
+restrictions, by denying admission of any pods that are trying to use internal knowledge of InterfaceID
+or Bitstream ID environment variables used by the prestart hook. 
+
+#### CRI-O prestart hook
+
+The [FPGA prestart CRI-O hook](cmd/fpga_crihook/README.md) is triggered by pod annotation set by the
+FPGA device plugin. It performs discovery of the requested FPGA function bitstream and then programs
```

Comment:
set by the FPGA admission controller?

Reply:
so s/set by the FPGA device plugin/set by the FPGA admission controller/ ?

I took the text directly from https://builders.intel.com/docs/networkbuilders/intel-device-plugins-for-kubernetes-appnote.pdf ....

Reply:
FPGA admission controller translates AFU and Region names into ids using [this mapping](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/master/deployments/fpga_admissionwebhook/mappings-collection.yaml), for example this [demo workload](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/master/demo/test-fpga-region.yaml) requests fpga.intel.com/arria10.dcp1.2-nlb3 function. Admission controller webhook translates this into an interfaceID/afuID pair: 69528db6eb31577a8c3668f9faa081f6: f7df405cbd7acf7222f144b0b93acd18.  Prestart hook uses this information to get a bitstream from bitstream storage (/srv/intel.com/fpga/69528db6eb31577a8c3668f9faa081f6/f7df405cbd7acf7222f144b0b93acd18.gbs) and program it into a device using fpga_tool utility installed by the initcontainer.

Reply:
OK, thx @bart0sh @mythi - but, you're going to have to guide me on what to write here.... what would you want to see in the docs if this was your first visit to the repo?
Details we can put into the specific plugins subdir docs, but we need some top level docs explaining what things are in the repo as a whole, what they roughly do, and where to find them :-)

Reply:
feel free to inject suggested wording and paragraphs in the comments here, and I can pick them up and fold them in.

Reply:
@rojkov can explain this in more details as he is an author of the webhook.

Reply:
No need to mention the admission controller here. The current wording is almost correct except one crucial inaccuracy: "The [FPGA prestart CRI-O hook](cmd/fpga_crihook/README.md) is triggered by **container** annotation set by the FPGA device plugin."
Pod annotation is a K8s-related thing.
Container annotation is what container runtimes deal with.

Reply:
updated. will push soon.